samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -0,0 +1,245 @@
+
+			base_bis_index_bitfield |= BIT(index);
+		}
+	}
+
+	bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
+
+	k_sem_give(&sem_base_received);
+}
+
+static void syncable_cb(struct bt_audio_broadcast_sink *sink, bool encrypted)
+{
+	if (encrypted) {
+		printk("Cannot sync to encrypted broadcast source\n");
+		return;
+	}
+
+	k_sem_give(&sem_syncable);
+}
+
+static void pa_sync_lost_cb(struct bt_audio_broadcast_sink *sink)
+{
+	if (broadcast_sink == NULL) {
+		printk("Unexpected PA sync lost\n");
+		return;
+	}
+
+	printk("Sink %p disconnected\n", sink);
+
+	broadcast_sink = NULL;
+
+	k_sem_give(&sem_pa_sync_lost);
+}
+
+static struct bt_audio_broadcast_sink_cb broadcast_sink_cbs = {
+	.scan_recv = scan_recv_cb,
+	.scan_term = scan_term_cb,
+	.base_recv = base_recv_cb,
+	.syncable = syncable_cb,
+	.pa_synced = pa_synced_cb,
+	.pa_sync_lost = pa_sync_lost_cb
+};
+
+static int init(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth enable printked (err %d)\n", err);
+		return err;
+	}
+
+	printk("Bluetooth initialized\n");
+
+	bt_audio_broadcast_sink_register_cb(&broadcast_sink_cbs);
+
+	for (size_t i = 0U; i < ARRAY_SIZE(streams); i++) {
+		streams[i].ops = &stream_ops;
+	}
+
+	return 0;
+}
+
+static void reset(void)
+{
+	bis_index_bitfield = 0U;
+
+	k_sem_reset(&sem_broadcaster_found);
+	k_sem_reset(&sem_pa_synced);
+	k_sem_reset(&sem_base_received);
+	k_sem_reset(&sem_syncable);
+	k_sem_reset(&sem_pa_sync_lost);
+
+	if (broadcast_sink != NULL) {
+		bt_audio_broadcast_sink_delete(broadcast_sink);
+		broadcast_sink = NULL;
+	}
+}
+
+void main(void)
+{
+	int err;
+
+	err = init();
+	if (err) {
+		printk("Init failed (err %d)\n", err);
+		return;
+	}
+
+	while (true) {
+		reset();
+
+		printk("Scanning for broadcast sources\n");
+		err = bt_audio_broadcast_sink_scan_start(BT_LE_SCAN_ACTIVE);
+		if (err != 0) {
+			printk("Unable to start scan for broadcast sources: %d\n",
+			       err);
+			return;
+		}
+
+		/* TODO: Update K_FOREVER with a sane value, and handle error */
+		err = k_sem_take(&sem_broadcaster_found, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_broadcaster_found timed out, resetting\n");
+			continue;
+		}
+		printk("Broadcast source found, waiting for PA sync\n");
+
+		k_sem_take(&sem_pa_synced, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_pa_synced timed out, resetting\n");
+			continue;
+		}
+		printk("Broadcast source PA synced, waiting for BASE\n");
+
+		k_sem_take(&sem_base_received, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_base_received timed out, resetting\n");
+			continue;
+		}
+		printk("BASE received, waiting for syncable\n");
+
+		k_sem_take(&sem_syncable, SEM_TIMEOUT);
+		if (err != 0) {
+			printk("sem_syncable timed out, resetting\n");
+			continue;
+		}
+
+		printk("Syncing to broadcast\n");
+		err = bt_audio_broadcast_sink_sync(broadcast_sink,
+						   bis_index_bitfield,
+						   streams,
+						   &preset_16_2_1.codec, NULL);
```

Comment:
I know it is not directly related to this change, but it is not obvious why an audio sink needs to provide a codec configuration to synchronize to a broadcast - normally it is the opposite that the sink is "told" by the broadcaster which codec to use.
Maybe add a few lines to the documentation about why this is needed, and from where the codec config should be obtained?
I actually think the codec parameter should be removed and the codec parameters from the big-info should be set by the host stack - if the provided codec config is not a 100% match to the one used on the broadcaster side it will not be able to decode the audio..

Reply:
Looking at the implementation, it doesn't look like the broadcast_sink needs a `codec` parameter anymore. I think this is a leftover from an older version where we probably needed to set a codec to avoid failing some check somewhere, but now that the roles have been split more, as well as improvements to the ISO API, I think we can remove this. 

Created https://github.com/zephyrproject-rtos/zephyr/issues/44455 for this